Xamarin.Essentials/Connectivity/Connectivity.android.cs

```diff
@@ -21,6 +22,38 @@ static void StartListeners()
             Platform.AppContext.RegisterReceiver(conectivityReceiver, new IntentFilter(ConnectivityManager.ConnectivityAction));
         }
 
+        static SignalStrength PlatformSignalStrength()
+        {
+            Permissions.EnsureDeclared(PermissionType.WifiState);
+            try
+            {
+                var wifiManager = Platform.WifiManager;
+                if (wifiManager == null)
+                    return SignalStrength.Unknown;
+
+                var info = wifiManager.ConnectionInfo;
+                if (info == null)
+                    return SignalStrength.None;
+
+                switch (WifiManager.CalculateSignalLevel(info.Rssi, 3))
+                {
+                    case 1:
+                        return SignalStrength.Weak;
+                    case 2:
+                        return SignalStrength.Fair;
+                    case 3:
+                        return SignalStrength.Strong;
+                    default:
+                        throw new InvalidOperationException("Exhaustive switch statement hit default value.");
```

Comment:
Maybe rather than throwing, since this comes from the platform and we have no control, just return unknown?

Reply:
I changed it on android as well as uwp to write a debug message and return unknown if it encounters an invalid value in the range provided